lib/mpool/src/mblock_file.c

```diff
@@ -794,6 +794,14 @@ mblock_file_alloc(struct mblock_file *mbfp, int mbidc, uint64_t *mbidv)
     mbid |= ((uint64_t)mbfp->mcid << MBID_MCID_SHIFT);
     mbid |= (block - 1);
 
+    if (flags & MPOOL_MBLOCK_PREALLOC) {
+        size_t mblocksz = mbfp->mblocksz;
+        int    rc;
+
+        rc = posix_fallocate(mbfp->fd, block_off(mbid, mblocksz), mblocksz);
+        ev(rc); /* advisory */
+    }
```

Comment:
Weird that mblocksz isn't an off_t...

Reply:
Why not fail if posix_fallocate() fails?

Reply:
The flag is documented as advisory, based on that it makes sense not to fail.

Reply:
> Weird that mblocksz isn't an off_t... 
You mean the "len" argument to the fallocate(), right? I incorrectly assumed that it's a size_t as it is a length from the start offset. Now I realize that it's an off_t. Will fix it.

> Also, you use uint32_t here for flags, and uint elsewhere... Might want to be consistent...
Will fix, thnx!

> Why not fail if posix_fallocate() fails?
We could fail the mpool_mblock_alloc(). I was contemplating on that and took the route of making the flag advisory as it is not going to cause incorrectness if fallocate fails. Please let me know if there's a hole in my thinking.

Reply:
No, was just thinking that if I pass that flag in and there's not enough space then things might fail later when I'm trying to fill the mblock.  So not exactly clear on what the flag buys me...???

Reply:
Right, ENOSPC will be caught later when writing to an mblock, however, if there are other concurrent deletes going on it might even succeed.

I think what the flag buys us is (1) Reduced FS metadata overhead as the allocation metadata need not be updated during mblock writes (2) Reduced FS fragmentation due to fixed size allocation and deallocations in the FS if cN sets this flag for majority of its allocations

Reply:
Sounds good!  So then, why have a flag, any reason not to always do it?

Reply:
Cool! Yeah, the presence of this flag helps us avoid space-amp for tiny kvsets. cN now has a knob and it won't set this flag if it knows that it's going to partially fill an mblock.

Reply:
Ah, right, forgot about that!!!  :-)